packages/ember-metal/lib/property_get.js

```diff
@@ -51,7 +51,7 @@ export function get(obj, keyName) {
   assert(`Get must be called with two arguments; an object and a property key`, arguments.length === 2);
   assert(`Cannot call get with '${keyName}' on an undefined object.`, obj !== undefined && obj !== null);
   assert(`The key provided to get must be a string, you passed ${keyName}`, typeof keyName === 'string');
-  assert(`'this' in paths is not supported`, !hasThis(keyName));
+  assert(`'this' in paths is not supported`, keyName.lastIndexOf('this.', 0) !== 0);
```

Comment:
I'm not sure we have `lastIndexOf` support on all the browsers we support:http://caniuse.com/#search=lastIndexOf

In theory we could use a helper function to choose the built-in or fallback to polyfill?